alicat/serial.py

```diff
@@ -132,21 +132,175 @@ def set_gas(self, gas, retries=2):
         """Set the gas type.
 
         Args:
-            gas: The gas type, as a string. Supported gas types are:
+            gas: The gas type, as a string or integer. Supported gas types by string are:
                 'Air', 'Ar', 'CH4', 'CO', 'CO2', 'C2H6', 'H2', 'He', 'N2',
                 'N2O', 'Ne', 'O2', 'C3H8', 'n-C4H10', 'C2H2', 'C2H4',
                 'i-C2H10', 'Kr', 'Xe', 'SF6', 'C-25', 'C-10', 'C-8', 'C-2',
                 'C-75', 'A-75', 'A-25', 'A1025', 'Star29', 'P-5'
+
+                For the full gas table, please see page 52 of the controller manual here:
+                https://documents.alicat.com/manuals/DOC-MANUAL-MC.pdf
+
+                Gas mixes may only be called by their mix number.
         """
         self._test_controller_open()
 
-        if gas not in self.gases:
-            raise ValueError("{} not supported!".format(gas))
-        command = '{addr}$${gas}\r'.format(addr=self.address,
-                                           gas=self.gases.index(gas))
+        if isinstance(gas, str) is False:
+            num = True
+        else:
+            num = False
```

Comment:
Replace with

```python
num = not isinstance(gas, str)
```

In the future, avoid using `is` this way. It doesn't quite mean what you think it means and it'll get you into trouble!

Also, in the future, determining if something is a string in python is harder than you'd expect. Python 2/3 handle it differently (`str` vs `basestring`) so it's better to check if number vs. not.

Reply:
Actually, looking at this more, I'd recommend breaking up into some private methods. Something like:

```python
def set_gas(self, gas)
    # if gas not in gases, etc at the front. Whatever's shared.
    if isinstance(gas, int):
        return self._set_gas_number(gas)
    else:
        return self._set_gas_name(gas)

def _set_gas_number(self, number):
    # write, read, check

def _set_gas_name(self, name):
    # write, read, check
```

Let me know if this makes sense to you!